### PR TITLE
Group Finder

### LINKED ISSUE #169: Groups: Group finder dashboard
## Acceptance Criteria

- [x] It should have a form matching current app
- [ ] It should list relevant articles below
- [x] Tapping "Lets go" navigates to search results

## TODO:

@Lepozepo need to make sure data queries/mutation dependencies are either implemented or issues are created for them.

### LINKED ISSUE #170: Groups: Group finder search results
## Acceptance Criteria

- [x] It should show search results in a list
- [x] Results should infinitely scroll
- [x] Should list search filters at top (this issue doesn't need to handle ability to modify search filters)

### LINKED ISSUE #172: Groups: Search finder filters
## Acceptance Criteria

- [x] It should have a text input that allows you to type text search filter
- [x] It should have a list of filters already selected, tapping them removes that filter
- [x] it should have dropdown menu of additional filters to select

### LINKED ISSUE #173: Groups: Group single page
## Acceptance Criteria

- [x] It should match design on current app
- [x] Only interactivity needed is to tab to open a contact form on web

### PR DESCRIPTION
**Pull request by conrad-vanl**
MVP Implementation of the group finder.

Fixes #169 
Fixes #170 
Fixes #172 
Fixes #173 

## Creator

- [ ] Build relevant tests if any apply
- [x] Ensure there are no new warnings
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer

## Reviewer

- [ ] Run `test` and make sure all tests pass
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review function and form on Web
- [ ] Review new test logic

### CODE REVIEW

File: src/@ui/NativeWebRouter/Link.js [hunk 1]
```diff
@@ -45,20 +45,23 @@ export default class Link extends Component {
```
Comment (Lepozepo) [inline src/@ui/NativeWebRouter/Link.js]: Not too sure how this happened. But when clicking on a link web errors with this:

screen shot 2018-01-03 at 10 01 17 am

It happens specifically when you click while waiting for group results to load. There may be something undefined while that's happening that should have a default?

File: src/@data/mediaPlayer/albumsQuery.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ export default gql`
   query getAlbums($limit: Int!, $skip: Int!) {
     library: content(channel: "newspring_albums", limit: $limit, skip: $skip) {
       id
-      entryId: id
```
[the remark this answers is not in the document]
Comment (conrad-vanl) [reply src/@data/mediaPlayer/albumsQuery.js]: Done in 5597ee9

I think that's right, as Apollo caches by `id` by default.

File: src/@ui/FeedView/FeedList.web.js [hunk 3]
```diff
@@ -50,14 +60,14 @@ class FeedList extends PureComponent {
       this.props.onEndReached();
```
[the remark this answers is not in the document]
Comment (conrad-vanl) [reply src/@ui/FeedView/FeedList.web.js]: Should be fixed in e151269f2f662039b295374da6e6f01bcfb260b8

**Comment by conrad-vanl** [comment]
@Lepozepo I believe I addressed the issues you pointed about above. Before we can merge:

- [ ] we need to fix the infinite scrolling issues
- [ ] I'd like @burkeshartsis's opinion of `@ui/Card`
- [ ] If opinion above is positive, I'll add tests for `Card` and refactor `FeedItemCard` to use the same components

File: src/@ui/NativeWebRouter/Link.js (hunk 1 above)
Comment (conrad-vanl) [reply src/@ui/NativeWebRouter/Link.js]: Fixed in 554796f

File: src/@ui/forms/GroupSearchForm/index.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      return props.onSubmit(query);
+    },
+  }),
+  setPropTypes({
+    setFieldValue: PropTypes.func,
+    setFieldTouched: PropTypes.func,
+    touched: PropTypes.shape({}),
+    errors: PropTypes.shape({}),
+    values: PropTypes.shape({}),
+    campuses: PropTypes.arrayOf(PropTypes.shape({})),
```
Comment (Lepozepo) [inline src/@ui/forms/GroupSearchForm/index.js]: We should set these shapes!

**Comment by Lepozepo** [comment]
@conrad-vanl I patched the graphql error, it should be ok (but brittle) now. Data patches continue [here](https://github.com/NewSpring/Apollos/issues/212)

**Pull request merged**